interface/forms/group_attendance/functions.php

```diff
@@ -0,0 +1,113 @@
+        $sqlBindArray = array();
+        array_push($sqlBindArray, $pc_aid, $pid, $gid, $pc_eventDate, $recurrspec, $pc_startTime, $converted_status);
+        sqlInsert($insert_sql, $sqlBindArray);
+    }
+}
+
+function insert_patient_encounter($pid, $gid, $group_encounter_date, $participantData, $pc_aid){
+    $select_sql = "SELECT id FROM form_encounter WHERE pid = ? AND external_id = ? AND pc_catid = ? AND date = ?; ";
+    $result = sqlStatement($select_sql, array($pid, $gid, 1000, $group_encounter_date));
+    $result_array = sqlFetchArray($result);
+    if($result_array){
+        $insert_sql = "UPDATE form_encounter SET reason = ? WHERE id = ?;";
+        sqlInsert($insert_sql, array($participantData['comment'], $result_array['pc_eid']));
+    }
+    else{
+        $insert_encounter_sql =
+            "INSERT INTO form_encounter (date, reason, pid, encounter, pc_catid, provider_id, external_id) ".
+            "VALUES (?, ?, ?, ?, 1000, ?, ?);";
+        $enc_id = generate_id();
+        $sqlBindArray = array();
+        array_push($sqlBindArray, $group_encounter_date, $participantData['comment'], $pid, $enc_id, $pc_aid, $gid);
+        $form_id = sqlInsert($insert_encounter_sql, $sqlBindArray);
+
+        global $userauthorized;
+
+        addForm($enc_id, "New Patient Encounter", $form_id, "newpatient",$pid, $userauthorized, $group_encounter_date,'','',NULL);
+
+    }
+}
+
+function get_appt_data($encounter_id){
+    $sql =
+        "SELECT ope.pc_aid, ope.pc_eventDate, ope.pc_startTime FROM form_groups_encounter as fge " .
+        "JOIN openemr_postcalendar_events as ope ON fge.appt_id = ope.pc_eid " .
+        "WHERE fge.encounter = ?;";
+    $result = sqlStatement($sql, array($encounter_id));
+    $result_array = sqlFetchArray($result);
+    return $result_array;
+}
+
+function get_group_encounter_data($encounter_id){
+    $sql = "SELECT date FROM form_groups_encounter WHERE encounter = ?";
+    $result = sqlStatement($sql, array($encounter_id));
+    $result_array = sqlFetchArray($result);
+    return $result_array;
+}
+
+function if_to_create_for_patient($status){
+    if($status == 20 || $status == 30 || $status == 40)
+        return true;
+    return false;
+}
+
+function convertStatus($status){
+    global $statuses_in_meeting_for_appt;
+    $converted_status = $statuses_in_meeting_for_appt[$status];
+    return $converted_status;
+}
+
+function jumpToEdit($form_id){
+    $url = "{$GLOBALS['rootdir']}/patient_file/encounter/view_form.php?formname=group_attendance&id=$form_id";
+    echo "\n<script language='Javascript'>top.restoreSession();window.location='$url';</script>\n";
+    exit;
+}
+
+function largest_id_plus_one($table){
+    $maxId = largest_id($table);
+    if ($maxId) {
+        $newid = $maxId + 1;
+    } else {
+        $newid = 1;
+    }
+    return $newid;
+}
+
+function largest_id($table){
+    $res = sqlStatement("SELECT MAX(id) as largestId FROM `" . $table . "`");
```

Comment:
should escape table, which can be done with:
``````
escape_table_name
``````
this function is included automatically in globals.php and can see function here:
https://github.com/openemr/openemr/blob/master/library/formdata.inc.php#L138-L148